src/ontology/OntologyInferenceAnnotatedTypeFactory.java

```diff
@@ -49,50 +43,6 @@ public OntologyInferenceAnnotatedTypeFactory(
         postInit();
     }
 
-    @Override
-    public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
-        return new OntologyInferenceQualifierHierarchy(factory);
-    }
-
-    public class OntologyInferenceQualifierHierarchy extends InferenceQualifierHierarchy {
-
-        public OntologyInferenceQualifierHierarchy(MultiGraphFactory multiGraphFactory) {
-            super(multiGraphFactory);
-        }
-
-        @Override
-        protected Set<AnnotationMirror> findBottoms(
-                Map<AnnotationMirror, Set<AnnotationMirror>> supertypes) {
-            Set<AnnotationMirror> newBottoms = super.findBottoms(supertypes);
-            newBottoms.remove(OntologyUtils.ONTOLOGY);
-            newBottoms.add(OntologyUtils.ONTOLOGY_BOTTOM);
-
-            // update supertypes
-            Set<AnnotationMirror> supertypesOfBtm = new HashSet<>();
-            supertypesOfBtm.add(OntologyUtils.ONTOLOGY_TOP);
-            supertypes.put(OntologyUtils.ONTOLOGY_BOTTOM, supertypesOfBtm);
-
-            return newBottoms;
-        }
-
-        @Override
-        protected void finish(
-                QualifierHierarchy qualHierarchy,
-                Map<AnnotationMirror, Set<AnnotationMirror>> fullMap,
-                Map<AnnotationMirror, AnnotationMirror> polyQualifiers,
-                Set<AnnotationMirror> tops,
-                Set<AnnotationMirror> bottoms,
-                Object... args) {
-            super.finish(qualHierarchy, fullMap, polyQualifiers, tops, bottoms, args);
-
-            // substitue ONTOLOGY with ONTOLOGY_TOP in fullMap
-            assert fullMap.containsKey(OntologyUtils.ONTOLOGY);
-            Set<AnnotationMirror> ontologyTopSupers = fullMap.get(OntologyUtils.ONTOLOGY);
-            fullMap.put(OntologyUtils.ONTOLOGY_TOP, ontologyTopSupers);
-            fullMap.remove(OntologyUtils.ONTOLOGY);
-        }
-    }
-
```

Comment:
Why don't we need  to override these methods?

Reply:
Because these are handled by the real type factory instead.